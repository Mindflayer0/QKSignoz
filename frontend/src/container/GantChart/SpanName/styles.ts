import { Typography } from 'antd';
import styled from 'styled-components';

export const Span = styled(Typography.Paragraph)`
	&&& {
		font-size: 0.75rem;
		margin: 0;
	}
`;

export const Service = styled(Typography)`
	&&& {
		color: #acacac;
		font-size: 0.75rem;
	}
`;

export const SpanWrapper = styled.div`
	display: flex;
	flex-direction: column;
	margin-left: 0.625rem;
	width: 10rem;
`;

export const Container = styled.div`
	display: flex;
	align-items: center;
	justify-content: flex-start;
`;
