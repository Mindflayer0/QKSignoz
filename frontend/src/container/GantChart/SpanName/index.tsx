import React from 'react';
import { Service, Span, SpanWrapper, Container } from './styles';

const SpanNameComponent = ({
	name,
	serviceName,
}: SpanNameComponent): JSX.Element => {
	return (
		<Container title={`${name} ${serviceName}`}>
			<SpanWrapper>
				<Span ellipsis>{name}</Span>
				<Service>{serviceName}</Service>
			</SpanWrapper>
		</Container>
	);
};

interface SpanNameComponent {
	name: string;
	serviceName: string;
}

export default SpanNameComponent;
