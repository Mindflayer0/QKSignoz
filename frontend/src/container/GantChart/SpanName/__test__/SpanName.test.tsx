import SpanName from '../index';
import { render } from '@testing-library/react';
import React from 'react';
import { expect } from '@jest/globals';

describe('Span Name', () => {
	it('render Span Name', async () => {
		const response = render(
			<SpanName
				{...{
					name: 'Same Name',
					serviceName: 'Sample Service Name',
				}}
			/>,
		);

		expect(response.container).toBeDefined();
		expect(response.container).toMatchSnapshot();
	});

	it('render Span Name', async () => {
		const response = render(
			<SpanName
				{...{
					name: 'Same Name',
					serviceName: 'Sample Service Name',
				}}
			/>,
		);

		const el = response.getByText('Sample Service Name');
		expect(el.nodeName).toEqual('ARTICLE');

		console.log(el);
	});
});
