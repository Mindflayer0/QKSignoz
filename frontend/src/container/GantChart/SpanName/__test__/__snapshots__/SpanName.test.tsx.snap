// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Span Name render Span Name 1`] = `
.c2.c2.c2 {
  font-size: 0.75rem;
  margin: 0;
}

.c3.c3.c3 {
  color: #acacac;
  font-size: 0.75rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 0.625rem;
  width: 10rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div>
  <div
    class="c0"
    title="Same Name Sample Service Name"
  >
    <div
      class="c1"
    >
      <div
        class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line c2"
      >
        Same Name
      </div>
      <article
        class="ant-typography c3"
      >
        Sample Service Name
      </article>
    </div>
  </div>
</div>
`;
